test/test_utils/__init__.py

```diff
@@ -836,21 +836,24 @@ def parse_canary_images(framework, region):
     customer_type_tag = f"-{customer_type}" if customer_type else ""
 
     version_regex = {
-        "tensorflow": rf"tf(-sagemaker)?{customer_type_tag}-(\d+.\d+)",
-        "mxnet": rf"mx(-sagemaker)?{customer_type_tag}-(\d+.\d+)",
-        "pytorch": rf"pt(-sagemaker)?{customer_type_tag}-(\d+.\d+)",
-        "huggingface_pytorch": r"hf-\S*pt(-sagemaker)?-(\d+.\d+)",
-        "huggingface_tensorflow": r"hf-\S*tf(-sagemaker)?-(\d+.\d+)",
-        "autogluon": r"ag(-sagemaker)?-(\d+.\d+)",
+        "tensorflow": rf"tf(-sagemaker)?{customer_type_tag}-(\d+.\d+)\S*(py\d+)",
+        "mxnet": rf"mx(-sagemaker)?{customer_type_tag}-(\d+.\d+)\S*(py\d+)",
+        "pytorch": rf"pt(-sagemaker)?{customer_type_tag}-(\d+.\d+)\S*(py\d+)",
+        "huggingface_pytorch": r"hf-\S*pt(-sagemaker)?-(\d+.\d+)\S*(py\d+)",
+        "huggingface_tensorflow": r"hf-\S*tf(-sagemaker)?-(\d+.\d+)\S*(py\d+)",
+        "autogluon": r"ag(-sagemaker)?-(\d+.\d+)\S*(py\d+)",
     }
 
     repo = git.Repo(os.getcwd(), search_parent_directories=True)
 
     versions_counter = {}
+    pre_populated_py3_version = {}
```

Comment:
this isn't always explicitly py3 right - the regex is general for python version